python/ray/serve/scripts.py

```diff
@@ -409,8 +409,25 @@ def run(
             import_attr(import_path), args_dict
         )
 
-    # Setting the runtime_env here will set defaults for the deployments.
-    ray.init(address=address, namespace=SERVE_NAMESPACE, runtime_env=final_runtime_env)
+    # Only initialize ray if it has not happened yet.
+    if not ray.is_initialized():
+        # Setting the runtime_env here will set defaults for the deployments.
+        ray.init(
+            address=address, namespace=SERVE_NAMESPACE, runtime_env=final_runtime_env
+        )
+
+    # Warning users the address they passed is different from the existing ray instance.
+    worker = ray._private.worker.global_worker
+    ray_address = worker.node.address
+    if address is not None and address != ray_address:
```

Comment:
would be ideal if we can use a public API instead of reaching into `global_worker.node`

Looks like `ray.get_runtime_context()` has a `gcs_address`, is that the same address?
https://docs.ray.io/en/latest/_modules/ray/runtime_context.html